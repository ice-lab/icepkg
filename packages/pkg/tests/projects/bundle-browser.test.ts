import { runProjectTest } from "./helper";

runProjectTest('bundle-browser', [{
  name: 'default',
  config: {
    transform: { formats: [] },
    bundle: {
      formats: ['esm', 'cjs']
    }
  }
}, {
  name: 'enable-browser',
  config: {
    transform: { formats: [] },
    bundle: {
      formats: ['esm', 'cjs'],
      browser: true
    }
  }
}])
