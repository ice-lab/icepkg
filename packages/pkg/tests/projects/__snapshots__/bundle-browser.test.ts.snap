// Vitest Snapshot v1

exports[`Run config default > cjs structure 1`] = `null`;

exports[`Run config default > dist structure 1`] = `
{
  "files": [
    {
      "name": "index.cjs.es5.production.js",
    },
    {
      "name": "index.esm.es5.production.js",
    },
  ],
  "name": "dist",
}
`;

exports[`Run config default > es2017 structure 1`] = `null`;

exports[`Run config default > esm structure 1`] = `null`;

exports[`Run config default > file content dist/index.cjs.es5.production.js 1`] = `
"\\"use strict\\";console.log(\\"module\\");
"
`;

exports[`Run config default > file content dist/index.esm.es5.production.js 1`] = `
"console.log(\\"module\\");
"
`;

exports[`Run config enable-browser > cjs structure 1`] = `null`;

exports[`Run config enable-browser > dist structure 1`] = `
{
  "files": [
    {
      "name": "index.cjs.es5.production.js",
    },
    {
      "name": "index.esm.es5.production.js",
    },
  ],
  "name": "dist",
}
`;

exports[`Run config enable-browser > es2017 structure 1`] = `null`;

exports[`Run config enable-browser > esm structure 1`] = `null`;

exports[`Run config enable-browser > file content dist/index.cjs.es5.production.js 1`] = `
"\\"use strict\\";console.log(\\"browser\\");
"
`;

exports[`Run config enable-browser > file content dist/index.esm.es5.production.js 1`] = `
"console.log(\\"browser\\");
"
`;
